config/config-api/src/main/java/com/thoughtworks/go/Deny.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go;
+
+import com.thoughtworks.go.config.*;
+import com.thoughtworks.go.domain.ConfigErrors;
+
+@ConfigTag("deny")
+public class Deny implements Directive {
+    @ConfigAttribute(value = "action", optional = false)
```

Comment:
Should we consider adding `@SkipParameterResolution` to all types where Param resolution is not needed?

Reply:
+1
Seems like a good idea, you will be able to decide where a secret parameter should not to be resolved.

Reply:
Not required as root SecretConfigs skips this.